server/src/main/java/org/elasticsearch/index/query/ExistsQueryBuilder.java

```diff
@@ -77,6 +77,18 @@ public String fieldName() {
         return this.fieldName;
     }
 
+    @Override
+    protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) throws IOException {
+        QueryShardContext context = queryShardContext.convertToShardContext();
+        if (context != null) {
+            MappedFieldType fieldType = context.fieldMapper(fieldName);
+            if (!fieldName.contains("*") && fieldType == null) {
```

Comment:
We should reuse the same logic as `newFilter` a couple lines below to resolve fields, in order to make sure to not break the behavior of `exists` queries. Can you extract how fields are resolved to a separate method, use it here, and rewrite to a `MatchNoneQueryBuilder` if none of the fields is mapped?

Reply:
Sure, updated the PR. Let me know if that works